lib/AST/Decl.cpp

```diff
@@ -6475,6 +6475,7 @@ bool ClassDecl::isForeignReferenceType() const {
   if (!clangRecordDecl)
     return false;
 
+  // importerImplPtr being nullpointer here is not expected to cause any issue
```

Comment:
The comment should explain why this is the case rather than just stating what is already in the code. I think it should rather say something like we do not want to trigger diagnostics from this call.